cellxgene_resources/curation_qa.ipynb

```diff
@@ -656,6 +657,129 @@
+    "    '''\n",
+    "    par_df = region_dict['par'][1]\n",
+    "    par_df['par_sum'] = par_df.sum(numeric_only=True, axis=1)\n",
+    "    nonpar_df = region_dict['nonpar'][1]\n",
+    "    nonpar_df['nonpar_sum'] = nonpar_df.sum(numeric_only=True, axis=1)\n",
+    "    nonpar_par_df = pd.merge(nonpar_df, par_df, how='left', left_index=True, right_index=True) \n",
+    "    nonpar_par_df['nonPAR_to_PAR'] = nonpar_par_df['nonpar_sum']/nonpar_par_df['par_sum']\n",
+    "    nonpar_par_df.rename(columns={'donor_id_x':'donor_id'}, inplace=True)\n",
+    "    nonpar_par_df['scRNAseq_sex'] = nonpar_par_df['nonPAR_to_PAR'].apply(assign_sex)\n",
+    "    return nonpar_par_df\n",
+    "\n",
+    "def check_donor_sex(nonpar_par_df, adata):\n",
+    "    '''\n",
+    "    Input: nonpar_par dataframe and original anndata object\n",
+    "    Output: print out of only the inconsistent donor's nonPAR_to_par ratio, scRNAseq-calculated sex, and author-labeled sex\n",
+    "    '''\n",
+    "    donors = adata.obs.value_counts(['donor_id', 'sex']).reset_index().rename(columns={\"index\": \"value\", 0: \"count\"})\n",
+    "    donors.drop(columns=['count'],inplace=True)\n",
+    "    check_donors = pd.merge(nonpar_par_df, donors, how='left', left_on='donor_id', right_on='donor_id')\n",
+    "    check_donors['mismatched'] = check_donors['scRNAseq_sex'].str.lower() != check_donors['sex'].str.lower()\n",
+    "    mismatches = check_donors[check_donors['mismatched'] == True]\n",
+    "    print(f\"Inconsistent Donor(s): \\n{mismatches[['donor_id', 'nonPAR_to_PAR', 'scRNAseq_sex', 'sex']]}\\n\")\n",
+    "    \n",
+    "def check_percent(par_adata,nonpar_adata,par_ids,nonpar_ids):\n",
+    "    '''\n",
+    "    Input: subset adatas and id lists\n",
+    "    Output: percent of genes found per region\n",
+    "    '''\n",
+    "    print(f\"% PAR genes found: {(par_adata.shape[1]/len(par_ids['gene_ids']))*100}\")\n",
+    "    print(f\"% Non-PAR genes found: {(nonpar_adata.shape[1]/len(nonpar_ids['gene_ids']))*100}\\n\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Run scRNA-seq sex analysis**<br>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#Get PAR & Non-PAR gene_ids from ref_file\n",
+    "ref_file = 'ref_files/gtf.tsv'\n",
+    "ref = pd.read_csv(ref_file, sep='\\t')\n",
+    "par_ids = ref[ref['gene_symbols'].isin(regions['PAR'])]\n",
+    "nonpar_ids = ref[ref['gene_symbols'].isin(regions['NONPAR'])]\n",
+    "\n",
+    "if adata.raw:\n",
+    "    #Subset PAR & Non-PAR regions from adata.raw.X\n",
+    "    par_adata = adata.raw[:,adata.raw.var.index.isin(par_ids['gene_ids'])]\n",
+    "    nonpar_adata = adata.raw[:,adata.raw.var.index.isin(nonpar_ids['gene_ids'])] \n",
+    "    \n",
+    "    #Check how many PAR & Non-PAR genes were found in dataset\n",
```

Comment:
Only the par_adata & nonpar_adata definitions are specific to the if/else logic. The rest (check_percent, region_counts, donor_df, check_donor_sex) can all be written just once after the if/else